src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea: Function,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
 
-}
+    const biggestSide = Math.max(this.a, this.b, this.c);
+    const sumOfSmallerSides = [this.a, this.b, this.c]
+      .reduce((sum, curr) => sum + curr) - biggestSide;
+
+    if (biggestSide >= sumOfSmallerSides) {
+      throw new Error('The longest side of the triangle must be less'
+      + 'than the sum of the other two');
+    }
+
+    if (this.a < 0 || this.b < 0 || this.c < 0
+      || biggestSide >= sumOfSmallerSides) {
+      throw new Error('Sides of the triangle must be bigger than 0');
```

Comment:
```suggestion
    if (this.a < 0 || this.b < 0 || this.c < 0) {
      throw new Error('Sides of the triangle must be bigger than 0');
```
Redundant check, you check for side biggest side size on line 23 already.